src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -76,7 +76,10 @@
 import com.puppycrawl.tools.checkstyle.filters.SuppressionFilter;
 import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
 
-public class CheckerTest extends BaseCheckTestSupport {
+/**
+ * @noinspection ClassWithTooManyDependencies
+ */
+public class CheckerTest extends AbstractModuleTestSupport {
```

Comment:
@subkrish I think it is because it is not allowed via https://github.com/checkstyle/checkstyle/blob/master/config/intellij-idea-inspections.xml#L2082 .
Please try adding it there.